File: languages/js/src/entity/key/totp_access_control.ts
export class TotpAccessControl {
  secret: string;
  secretQr: string;
  recoveryCodes: string[];

  constructor(secret: string, secretQr: string, recoveryCodes: string[]) {
    this.secret = secret;
    this.secretQr = secretQr;
    this.recoveryCodes = recoveryCodes;
  }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/key/TotpAccessControl.java
package com.bloock.sdk.entity.key;

import java.util.List;

public class TotpAccessControl {
  String secret;
  String secretQr;
  List<String> recoveryCodes;

  public TotpAccessControl(String secret, String secretQr, List<String> recoveryCodes) {
    this.secret = secret;
    this.secretQr = secretQr;
    this.recoveryCodes = recoveryCodes;
  }

  public String getSecret() {
    return secret;
  }

  public String getSecretQr() {
    return secretQr;
  }

  public List<String> getRecoveryCodes() {
    return recoveryCodes;
  }
}
